src/Rover.cpp

```diff
@@ -94,32 +95,82 @@ void parseCommandLine(int argc, char** argv) {
 			  "LOG_WARN, LOG_ERROR)")
 		.default_value(std::string("LOG_INFO"))
 		.action([](const std::string& value) {
-			std::unordered_map<std::string, int> allowed{{"LOG_TRACE", LOG_TRACE},
-														 {"LOG_DEBUG", LOG_DEBUG},
-														 {"LOG_INFO", LOG_INFO},
-														 {"LOG_WARN", LOG_WARN},
-														 {"LOG_ERROR", LOG_ERROR}};
+			std::unordered_map<std::string, int> allowed{
+				{"LOG_ERROR", loguru::Verbosity_ERROR},
+				{"LOG_WARN", loguru::Verbosity_WARNING},
+				{"LOG_INFO", loguru::Verbosity_INFO},
+				{"LOG_DEBUG", 2},
+				{"LOG_TRACE", 1},
+			};
 
 			if (allowed.find(value) != allowed.end()) {
-				setLogLevel(allowed[value]);
-				return value;
+				loguru::g_stderr_verbosity = allowed[value];
 			}
 
 			throw std::runtime_error("Invalid log level " + value);
 		});
 
 	program.add_argument("-nc", "--no-colors")
 		.help("disables colors in console logging")
-		.action([&](const auto&) { setColorsEnabled(false); })
+		.action([&](const auto&) { loguru::g_colorlogtostderr = false; })
 		.nargs(0);
 
 	try {
+		loguru::init(argc, argv);
+		namespace fs = std::filesystem;
+
+		const int LOG_LIFESPAN = 604800; // 7 days
+		try {
+			for (const auto& entry : fs::directory_iterator(fs::current_path())) {
+				std::cout << entry.path().extension();
+				if (entry.path().extension() == ".log" && entry.path().stem() != "latest") {
+					std::string dateString = entry.path().stem();
+
+					// Extract components from the date string
+					int year, month, day, hour, minute, second;
+					sscanf(dateString.c_str(), "%4d%2d%2d_%2d%2d%2d", &year, &month, &day,
+						   &hour, &minute, &second);
+
+					// Create a tm structure
+					std::tm tm_struct = {};
+					tm_struct.tm_year = year - 1900;
+					tm_struct.tm_mon = month - 1;
+					tm_struct.tm_mday = day;
+					tm_struct.tm_hour = hour;
+					tm_struct.tm_min = minute;
+					tm_struct.tm_sec = second;
+
+					// Current time
+					std::time_t unixTime = std::mktime(&tm_struct);
+
+					// Delete log if it's older than LOG_LIFESPAN
+					if (std::chrono::system_clock::to_time_t(
+							std::chrono::system_clock::now()) -
+							unixTime >
+						LOG_LIFESPAN) {
```

Comment:
`std::time_t` has very little guarantees, i.e. it's not guaranteed to be in seconds, and it's technically not even guaranteed to count up. Let's prefer other time representations, so we could do this instead:

```c++
if (std::chrono::system_clock::now() - std::chrono::from_time_t(unixTime) > LOG_LIFESPAN) { ... }
```